src/ima-state-machine.js

```diff
@@ -150,10 +150,10 @@ function onAdLoaded(options: Object, adEvent: any): void {
  */
 function onAdStarted(options: Object, adEvent: any): void {
   this.logger.debug(adEvent.type.toUpperCase());
-  let ad = adEvent.getAd();
+  this._currentAd = adEvent.getAd();
```

Comment:
we used to declare class properties at the head class. as you did in the other file

Reply:
oh?
this is an assignment and not a declaration.

Reply:
where the declaration?